crates/ra_ide/src/lib.rs

```diff
@@ -511,7 +511,7 @@ impl Analysis {
     }
 
     /// Performs an operation on that may be Canceled.
-    fn with_db<F: FnOnce(&RootDatabase) -> T + std::panic::UnwindSafe, T>(
+    pub fn with_db<F: FnOnce(&RootDatabase) -> T + std::panic::UnwindSafe, T>(
```

Comment:
can't you use `AnalysisHost::raw_database()` if necessary?

Reply:
I can, but I wanted the `cancel` ability and I felt this is the better way.

Reply:
Looks like I didn't need `with_db` and was okay with `raw_database()`. I can make this private back again if you guys want.

Reply:
r=me with `with_db` kept private. 

bors d+